src/filters/authority.rs

```diff
@@ -0,0 +1,92 @@
+//! Authority (aka `Host` header) filter
+//!
+use crate::filter::{filter_fn_one, Filter, One};
+use crate::reject::{self, Rejection};
+use futures::future;
+pub use http::uri::Authority;
+use std::str::FromStr;
+
+/// Creates a `Filter` that requires a specific authority (target server's
+/// host and port) in the request.
+///
+/// # Example
+///
+/// ```
+/// use warp::{Filter, authority::Authority};
+///
+/// let multihost =
+///     warp::authority("foo.com").map(|| "you've reached foo.com")
+///     .or(warp::authority("bar.com").map(|| "you've reached bar.com"));
+/// ```
+pub fn authority(expected: &str) -> impl Filter<Extract = (), Error = Rejection> + Clone {
```

Comment:
Since this filter requires a match, perhaps the name should indicate that. What these options: `required`, `guard`, or `exact` (also used in `warp::header::exact`)?

Reply:
Ok, I'll rename it to `exact` to match `header::exact` and remove the `warp::authority` altogether (so you either have to use `::exact` or `::optional`)